packages/opentelemetry-sdk-trace-web/src/utils.ts

```diff
@@ -125,6 +129,154 @@ export function addSpanNetworkEvents(
   }
 }
 
+function _getBodyNonDestructively(body: ReadableStream) {
+  // can't read a ReadableStream without destroying it
+  // but we CAN pipe it through and return a new ReadableStream
+
+  // some (older) platforms don't expose the pipeThrough method and in that scenario, we're out of luck;
+  //   there's no way to read the stream without consuming it.
+  if (!body.pipeThrough) {
+    return {
+      body,
+      length: Promise.resolve(undefined),
+    };
+  }
```

Comment:
I'm not sure whether a diagnostic or console.warning would be appropriate here (or what's generally preferred in the project for surfacing these kind of things to users), but it might be helpful to include some sort of information to help users diagnose why the length is missing. (Though I understand this would probably be a pretty rare occurrence given the combination of old platform + using readablestream + otel).